src/transformers/tokenization_utils_base.py

```diff
@@ -727,9 +727,9 @@ def convert_to_tensors(
 
                     self[key] = tensor
             except:  # noqa E722
-                if key == "overflowing_tokens":
+                if key == "overflowing_tokens" or key == "input_ids" or key == "attention_mask":
```

Comment:
Please do not touch this part of the test or the error raised. You can dd the information on the key in the error raised at line 736.

Reply:
I am trying to formulate an explanation to add at line 736 but I'm led back to saying what is already written at line 731. The reason that excessive nesting is an issue at collating is that even with padding the features are tokenized at different lengths. This is due to the additional wrapping of the actual feature list like so: `[[1,2]]` and `[[1,2,3]]`. They both share the shape `(1, ` but after that they are different shapes so what `np.array` does is: `[[ list ]]` and `[[ list ]]` thus returning shape`(1, 1)`.

That is why I added the code at 730 because it technically is the perfect explanation at this point in the process.